cpp/src/arrow/compute/api_aggregate.h

```diff
@@ -480,6 +480,12 @@ struct ARROW_EXPORT Aggregate {
 
   /// options for the aggregation function
   const FunctionOptions* options;
```

Comment:
We should convert this to `std::unique_ptr<FunctionOptions>` but I don't mind if we leave that for a follow-up PR.

Reply:
Follow-up PR created: https://issues.apache.org/jira/browse/ARROW-16686

Reply:
Why should it be `unique_ptr`? It's not obvious there's a need for ownership here; also, the caller might want to keep ownership as well.